libparsec/crates/platform_device_loader/src/web/mod.rs

```diff
@@ -56,34 +59,52 @@ pub async fn save_device(
         .map_err(Into::into)
 }
 
-pub async fn change_authentication(
+pub async fn update_device(
     current_access: &DeviceAccessStrategy,
     new_access: &DeviceAccessStrategy,
-) -> Result<AvailableDevice, crate::ChangeAuthentificationError> {
+    overwrite_server_addr: Option<ParsecAddr>,
+) -> Result<(AvailableDevice, ParsecAddr), UpdateDeviceError> {
     let storage = Storage::new().inspect_err(|e| {
         log::error!("Failed to access storage: {e}");
     })?;
-    let (device, created_on) = storage.load_device(current_access)?;
+
+    let (mut device, created_on) = storage.load_device(current_access)?;
+
+    let old_server_addr = ParsecAddr::new(
+        device.organization_addr.hostname().to_owned(),
+        Some(device.organization_addr.port()),
+        device.organization_addr.use_ssl(),
+    );
+    if let Some(overwrite_server_addr) = overwrite_server_addr {
+        Arc::make_mut(&mut device).organization_addr = ParsecOrganizationAddr::new(
+            overwrite_server_addr,
+            device.organization_addr.organization_id().to_owned(),
+            device.organization_addr.root_verify_key().to_owned(),
+        );
+    }
+
     let available_device = storage.save_device(new_access, &device, created_on)?;
 
     let key_file = current_access.key_file();
     let new_key_file = new_access.key_file();
 
     if key_file != new_key_file {
-        storage.remove_device(key_file)?;
+        if let Err(err) = storage.remove_device(key_file) {
+            log::warn!("Cannot remove old key file {key_file:?}: {err}");
+        }
     }
 
-    Ok(available_device)
+    Ok((available_device, old_server_addr))
```

Comment:
The old_server_address doesn't seem to be used anywhere, is it needed?

Reply:
This is used in the CLI to display the old address before overwritting it